### PR TITLE
fix some oss-fuzz bugs

### PR DESCRIPTION
**Pull request by chenhuitao**
fix some oss-fuzz bugs: 10419, 10427, 10421, 10422, 10425, 10426, 10460.

### CODE REVIEW

**Comment by aquynh** [comment]
@stephengroat could you look at the issue reported by Appveyor?

File: qemu/translate-all.c [hunk 1]
```diff
@@ -1451,9 +1451,13 @@ void tb_invalidate_phys_page_fast(struct uc_struct* uc, tb_page_addr_t start, in
     if (p->code_bitmap) {
         unsigned int nr;
         unsigned long b;
+        unsigned int x;
 
         nr = start & ~TARGET_PAGE_MASK;
-        b = p->code_bitmap[BIT_WORD(nr)] >> (nr & (BITS_PER_LONG - 1));
+        x = nr & (BITS_PER_LONG - 1);
```
Comment (aquynh) [inline qemu/translate-all.c]: how about just 1 line: `b = p->code_bitmap[BIT_WORD(nr)] >> ((nr & (BITS_PER_LONG - 1)) & 0x1f);`

Comment (chenhuitao) [reply qemu/translate-all.c]: O.K.

File: qemu/target-sparc/translate.c [hunk 2]
```diff
@@ -90,8 +90,9 @@ typedef struct {
 
 static int sign_extend(int x, int len)
 {
+    int ret;
```
Comment (aquynh) [inline qemu/target-sparc/translate.c]: this var should be removed

Comment (chenhuitao) [reply qemu/target-sparc/translate.c]: right.

**Comment by aquynh** [comment]
merged, thanks!

**Pull request merged**